lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,111 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var fetchOnce = null,
+            fetchNotes,
+            renderNotes,
+            _requestsData = null,
+            storeNotesRequestData,
+            SearchRequestsData = [],
+            waitUntilAllRequestsStored;
+
+        waitUntilAllRequestsStored = function (maxTimeout) {
```

Comment:
A simple polling to check if all the HTML components on a unit has stored their data.